File: homeworks/homework1/22002105_Gokmen_AhmetBerke.py
# Declare/create an empty list
empty_l = []
print("empty list:", empty_l)

# Initialize a list with some values
l = ['Apple', 'Avocado', 'Cherries', 'Dragonfruit', 'Grapefruit', 'Grapes', 'Hazelnut', 'Kiwi', 'Lime', 'Nectarines']
print("list:", l)

# Check if the list is empty or not
is_empty = len(l) == 0
print("is_empty:", is_empty)

# Add a new element to a list
l.append("Raspberries") # add Raspberries
l += ["Tangelo"] # add Cherries
l.extend(["Watermelon"]) # add Watermelon
print("list after adding items:", l)

# Check if a particular element exists in the list
does_watermelon_exist = l.count("Watermelon") != 0 # basic way
does_pomegranate_exist = len(list(filter(lambda e: e == "Pomegranate", l))) != 0 # a more complicated way
print(f"does_watermelon_exist: {does_watermelon_exist}, does_pomegranate_exist {does_pomegranate_exist}")

# Remove a particular element from the list
l.pop(len(l) - 1) # remove element at index, in this case the last element
l = list(filter(lambda e: e != "Cherries", l)) # remove Cherries from the array, a little bit more complicated
l = [e for e in l if e != "Raspberries"] # with list comprehensions
print("list after removing items:", l)

# Get the head and the tail of a list
head1, *tail1 = l
head2, tail2 = l[0], l[1:len(l)+1] # another way...
print(f"head: {head1}, tail: {tail1}")

# Print all of the elements in the list
list(map(lambda e: print(e), l)) # pythonic way of foreach

File: homeworks/homework1/22002105_Gokmen_AhmetBerke.dart
void main() {
  // Declare/create an empty list
  List<String> emptyList = List<String>.empty();
  print("emptyList: $emptyList");

  // Initialize a list with some values
  List<String> list = [
    'Apple', 
    'Avocado', 
    'Cherries', 
    'Dragonfruit', 
    'Grapefruit', 
    'Grapes', 
    'Hazelnut', 
    'Kiwi', 
    'Lime', 
    'Nectarines', 
    'Raspberries'
  ];
  print("list: $list");

  // Check if the list is empty or not
  bool is_empty = list.isEmpty;
  print("is_empty: $is_empty");

  // Add a new element to a list
  list.add("Tangelo");
  list.add("Watermelon");
  print("list after adding items: $list");

  // Check if a particular element exists in the list
  bool doesMelonExist = list.contains("Melon"); // basic way
  bool doesWatermelonExist = !list.where((e) => e == "Watermelon").isEmpty; // complicated way
  print("doesMelonExist: $doesMelonExist doesWatermelonExist: $doesWatermelonExist");

  // Remove a particular element from the list
  list = list
      .where((e) => e != "Tangelo")
      .toList(); // remove Tangelo with filtering
  list.remove("Kiwi"); // remove Kiwi with native api
  list.removeAt(0); // remove item at index 0
  print("list after removing items: $list");

  // Get the head and the tail of a list
  String head = list[0];
  List<String> tail = list.sublist(1);
  print("head: $head, tail: $tail");

  //Print all of the elements in the list
  print("list:");
  list.forEach((e) => print(e));
}
